### PR TITLE
Azure CLI 2.12.0

### PR DESCRIPTION
**Pull request by azclibot**
The latest release of the Azure CLI

### CODE REVIEW

File: Formula/azure-cli.rb
Comment (Rylan12) [inline Formula/azure-cli.rb:16]: The `livecheck` block should not be removed.

File: Formula/azure-cli.rb
Comment (Rylan12) [inline Formula/azure-cli.rb:65]: Just want to verify that this downgrade is intentional.

File: Formula/azure-cli.rb
Comment (Rylan12) [inline Formula/azure-cli.rb:340]: Unless 3.0.0rc1 is needed to fix a security vulnerability we shouldn't have a release candidate resource. Can we stick with 2.2.0?

**Review by Rylan12 — CHANGES_REQUESTED** [review]
@azclibot it looks like there might be a minor issue with the release bot; the `livecheck` block should not be removed.

Also, a few questions/clarifications on resources:

File: Formula/azure-cli.rb
Comment (bim-msft) [reply Formula/azure-cli.rb:65]: @Rylan12 Yes, this is intentional, we locked the version for this release.

File: Formula/azure-cli.rb
Comment (bim-msft) [reply Formula/azure-cli.rb:340]: @Juliehzl Can you confirm this? The version was bumped from here: https://github.com/Azure/azure-cli/pull/15177

File: Formula/azure-cli.rb
Comment (bim-msft) [reply Formula/azure-cli.rb]: Added again. Thanks.

File: Formula/azure-cli.rb
Comment (Juliehzl) [reply Formula/azure-cli.rb:345]: @rajsell could you help confirm if the new package fix security vulnerability?

Comment (Juliehzl) [reply Formula/azure-cli.rb:345]: By the way, could package be beta version? If not, we will try to avoid such usage in the future.

Comment (SMillerDev) [reply Formula/azure-cli.rb:345]: https://docs.brew.sh/Acceptable-Formulae#stable-versions we're a little more lenient with resources, but in general we like the software we ship to be stable (and I bet your customers do to)

Comment (fengzhou-msft) [reply Formula/azure-cli.rb:345]: @Rylan12 does @Juliehzl's reply sounds reasonable to you? Can we merge this PR to unblock our customers first?

Comment (Rylan12) [reply Formula/azure-cli.rb:345]: I'll defer to Sean for this. @SMillerDev you had indicated to me that we only wanted release candidate resources that fix vulnerabilities. That doesn't seem the be the case here, but staying with the last stable release doesn't seem to be an option. Thoughts?

Comment (SMillerDev) [reply Formula/azure-cli.rb:345]: I'm fine with using this, but please try and use stable versions in the future.

**Review by Rylan12 — APPROVED** [review]
LGTM. Thanks to everyone for your responses!

We do ask that you rely on stable versions for resources in the future, though.

**Comment by fengzhou-msft** [comment]
@Rylan12 @SMillerDev thanks for helping out and the clarification. The rc versions of Azure SDKs usually contain the stable APIs plus some preview APIs and in Azure CLI, we will tag the commands/parameters that use preview APIs as preview and prompt a warning to users when they use them.

While we will take an eye on this and use stable versions as possible as we can, it may require some time to completely change to stable versions.

**Pull request closed without merging**
